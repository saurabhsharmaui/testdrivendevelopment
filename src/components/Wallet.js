import React, { Component } from 'react';
import { deposit, withdrawal } from '../actions/balance';
import { connect } from 'react-redux';

export class Wallet extends Component {
    constructor() {
        super();
        this.state = {
            balance: undefined
        }
    }
    updateBalance = event => {
        this.setState({ balance: parseInt(event.target.value, 10) });
    }
    deposit = () => {
        this.props.deposit(this.state.balance);
    }
    withdraw = () => {
        this.props.withdrawal(this.state.balance);
    }
    render() {
        return (
            <div>
                <h3 className="balance">Wallet Balance:{this.props.balance}</h3>
                <br />
                <input className="input-wallet" onChange={this.updateBalance}></input>
                <button className="btn-deposit" onClick={this.deposit}>Deposit</button>
                <button className="btn-withdraw" onClick={this.withdraw}>WithDraw</button>
            </div>
        )
    }
}

export default connect(state => {
    return {
        balance: state.balance
    }
}, { deposit, withdrawal })(Wallet);