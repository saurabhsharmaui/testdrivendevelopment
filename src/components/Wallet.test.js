import React from 'react';
import { shallow } from 'enzyme';
import { Wallet } from './Wallet';

describe('Wallet', () => {
    const mockDeposite = jest.fn();
    const mockWithdraw = jest.fn();
    const props = { balance: 20, deposit: mockDeposite, withdrawal: mockWithdraw };


    const wallet = shallow(<Wallet  {...props} />);
    it('render properly', () => {
        expect(wallet).toMatchSnapshot();
    });
    it('display the balance from props', () => {
        expect(wallet.find('.balance').text()).toEqual('Wallet Balance:20');
    });
    it('creates an input to deposit into or withdraw from the balance', () => {
        expect(wallet.find('.input-wallet').exists()).toBe(true);
    });
    describe('when the user types into the waller input', () => {
        const userBalance = '25';
        beforeEach(() => {
            wallet.find('.input-wallet')
                .simulate('change', { target: { value: userBalance } })
        });
        it('updates the local wallet balance in state and converts it to a number', () => {
            expect(wallet.state().balance).toEqual(parseInt(userBalance, 10));
        });
        describe('and the user wants to maka a deposite', () => {
            beforeEach(() =>
                wallet.find('.btn-deposit').simulate('click')
            );
            it('dispatches the deposit() it rececives from props with locak balance', () => {
                expect(mockDeposite).toHaveBeenCalledWith(parseInt(userBalance, 10));
            });
        });
        describe('and the user wants to maka a withdraw', () => {
            beforeEach(() =>
                wallet.find('.btn-withdraw').simulate('click')
            );
            it('dispatches the withdraw() it rececives from props with locak balance', () => {
                expect(mockWithdraw).toHaveBeenCalledWith(parseInt(userBalance, 10));
            });
        });
    });



});

