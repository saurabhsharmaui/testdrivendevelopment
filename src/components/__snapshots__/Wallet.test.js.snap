// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet render properly 1`] = `ShallowWrapper {}`;
